core/channel.go

```diff
@@ -110,34 +110,22 @@ type Channel struct {
 }
 
 func (c *Connection) newChannel(reqChanBufSize, replyChanBufSize int) (*Channel, error) {
-	// create new channel
-	channel := &Channel{
-		conn:                c,
-		msgCodec:            c.codec,
-		msgIdentifier:       c,
-		reqChan:             make(chan *vppRequest, reqChanBufSize),
-		replyChan:           make(chan *vppReply, replyChanBufSize),
-		replyTimeout:        DefaultReplyTimeout,
-		receiveReplyTimeout: ReplyChannelTimeout,
+	// get a channel from the pool
+	channel := c.channelPool.Get()
+	if channel == nil {
+		return nil, errors.New("all channel IDs are in use")
+	}
+	if len(channel.reqChan) != reqChanBufSize {
```

Comment:
I believe that `len` function will return how many objects are in the channel. What you wanted is `cap` to show the capacity of the channel - buffer size.

-> https://go.dev/play/p/bB6Gwx_p-jj

Reply:
You are correct about cap.  Also realized in the course of thinking about it that we may have 'lingering' items in the channel.reqChan or channel.replyChan and so drained them just in case.